src/FS.Embedded/Manifest/ManifestRootDirectory.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class ManifestRootDirectory : ManifestDirectory
+    {
+        public ManifestRootDirectory(ManifestEntry[] children)
+            : base(null, children)
+        {
+            SetParent(ManifestSinkDirectory.Instance);
+        }
+
+        public override ManifestDirectory ToRootDirectory() => this;
```

Comment:
Does this type need to account for path navigation outside the root?

Reply:
It does automatically by virtue of the parent being a sink directory